File: Solutions/[17]_1.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int[] data = {500, 100, 50, 10, 5, 1};
		int changes, result = 0;
		
		changes = sc.nextInt();
		changes = 1000 - changes;
		for (int i = 0; i < 6; i++) {
			result += changes / data[i];
			changes %= data[i];
		}
		
		System.out.println(result);
	}

}

File: Solutions/[17]_1.cpp
#include <bits/stdc++.h>

using namespace std;

int data[6] = {500, 100, 50, 10, 5, 1};
int changes, result;

int main(void) {
	cin >> changes;
	changes = 1000 - changes;
	for (int i = 0; i < 6; i++) {
		result += changes / data[i];
		changes %= data[i];
	}
	cout << result;
}
